src/main/scala/gitbucket/core/model/WikiActivityActionInfo.scala

```diff
@@ -0,0 +1,82 @@
+package gitbucket.core.model
+
+import java.util.UUID
+
+import gitbucket.core.model.Profile.currentDate
+
+sealed trait WikiActivityActionInfo {
```

Comment:
By moving all the logic related with activity information into this sealed trait, I think we can enforce a better separation of concerns; we no longer pollute the `ActivityService` with the details of log formation, since we now delegate this responsibility to this trait and its subtypes.

Reply:
Do you intend all activity log generation currently implemented in `ActivityService` should be migrated to this way? I think it's better to migrate all activity logging methods at once if we introduce this than applying to only wiki activity logging and that refactoring might have to be another pull request.

Reply:
I think this method of delegating the log formatting to the data types themselves might be a better idea than having a lot of very specific functions in `ActivityService`. 

I'd be happy to perform the refactoring myself. Maybe I can do it in a PR separate from this one and merge it in first and we can play these changes on top of it?

Reply:
> I'd be happy to perform the refactoring myself. Maybe I can do it in a PR separate from this one and merge it in first and we can play these changes on top of it?

That sounds great.

> I tried to create an issue for this work but it looks like I don't have the necessary permissions

Hmm... That's weird. Everyone should be able to create an issue. Anyway, you already created a pull request #2526. Let me take a look.